### PR TITLE
Support packages that are not at the root of a git repository.

### PR DESCRIPTION
**Pull request by mbjones**
This fix adds 'discover=T' to git2r:repository() calls in order to allow devtools to locate the repository root directory when the package is in a subdirectory of the overall git repository.  This change should address issue #845.  I have tested this on Mac OS X under R 3.2.0.

I encountered this problem when trying to release() a package that was not at the root of its git repository, and the git_uncommitted() function threw an error as described in #845. After the fix, git_uncommitted() correctly detects the repository root.

### CODE REVIEW

File: R/git.R [hunk 1]
```diff
@@ -3,19 +3,19 @@ uses_git <- function(path = ".") {
 }
 
 git_sha1 <- function(n = 10, path = ".") {
-  r <- git2r::repository(path)
+  r <- git2r::repository(path, discover=T)
```
Comment (hadley) [inline R/git.R]: Could you please use `discover = TRUE`?

**Comment by hadley** [comment]
Thanks - could you please also add a bullet point to NEWS?

**Comment by mbjones** [comment]
Yes, I'll change to use 'TRUE', and add a bullet to news in a revised PR.  More soon...

**Comment by mbjones** [comment]
@hadley I made the changes you requested, and tests passed on my local Mac deployment, as well as in Travis.  I'm not clear on your development workflow -- do you want PRs to fix the codecov issues?  Or does this look good now?

**Comment by hadley** [comment]
codecov is mostly informatively - I'm still experimenting with it.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**